# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/rifkiay/flyrank-ml-internship-starter/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

**Lane: Refresh / Content Opportunity Scoring**

I'm picking this lane because the output is a ranked list a team can act on directly. With far more pages than any team could manually review, a ranked list is the most useful format — it tells someone exactly where to start.

In [1]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Decision:** which pages should a reviewer check first, since they can't review every page.

**Who acts:** a content reviewer.

**Action:** they open the top pages on the list and decide what to do (refresh, expand, protect, prune, or leave it).

**Cost of a wrong call:**
- Flagging a healthy page = wasted reviewer time.
- Missing a real problem page = it keeps losing traffic unnoticed.

**Why data helps:** too many pages to check by hand, so a score can rank them faster than a person can.


In [2]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [3]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd
from pathlib import Path

ROOT = Path.cwd().parent.parent

df = pd.read_csv(ROOT / "data/raw/content_refresh_anonymized.csv")

declining_with_demand = df[(df["trend_direction"] == "down") & (df["impressions_90d"] >= 100)]
print("Declining + demand:", len(declining_with_demand), "out of", len(df), "pages")

stale_visible = df[(df["days_since_last_update"] >= 180) & (df["impressions_90d"] >= 500)]
print("Stale + visible:", len(stale_visible))

print(df["trend_direction"].value_counts())

Declining + demand: 13152 out of 30000 pages
Stale + visible: 17
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


Out of 30,000 pages, 16,262 (about 54%) are trending down — more than the 4,388 trending up. Of those declining, 13,152 still have real search demand (100+ impressions in 90 days), which is a large enough group to make ranking them worth building over the next 7 weeks.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

**What I can say:** patterns I actually observe in the data, like how many pages are declining while still getting search demand — these are observed facts, not predictions.

**What I can't say:** that a refresh will fix a page (that needs a real experiment, not just this data), or that my ranking is always correct — it's a tool to help prioritize review, not a guarantee.

In [4]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [✅] Every section above is filled — markdown thinking AND the code that backs it
- [✅] The notebook runs top to bottom with no errors (Runtime → Run all)
- [✅] No client names, URLs, or private queries anywhere
- [✅] My claims use careful words: observed, measured, directional, decision-support
- [✅] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.